geomstats/sphere.py

```diff
@@ -0,0 +1,114 @@
+"""
+Computations on the n-dimensional sphere
+embedded in the (n+1)-dimensional Euclidean space.
+"""
+
+import numpy as np
+
+EPSILON = 1e-6
+
+
+def projection_to_tangent_space(ref_point, vector):
+    """
+    Project the vector v onto the tangent space:
+    T_{ref_point} S = {w | scal(w, ref_point) = 0}
+    """
+    sq_norm = np.dot(ref_point, ref_point)
+    tangent_vec = vector - np.dot(ref_point, vector) / sq_norm * ref_point
+    return tangent_vec
+
+
+def riemannian_exp(ref_point, vector, epsilon=EPSILON):
+    """
+    Compute the Riemannian exponential at point ref_point
+    of tangent vector tangent_vec wrt the metric obtained by
+    embedding of the n-dimensional sphere
+    in the (n+1)-dimensional euclidean space.
+
+    This gives a point on the n-dimensional sphere.
+
+    :param ref_point: a point on the n-dimensional sphere
+    :param vector: (n+1)-dimensional vector
+    :returns riem_exp: a point on the n-dimensional sphere
+    """
+    tangent_vec = projection_to_tangent_space(ref_point, vector)
+    norm_tangent_vec = np.linalg.norm(tangent_vec)
+
+    if norm_tangent_vec < epsilon:
```

Comment:
can you explain a little bit waht happens here? I assume this is a taylor expansion of the sine/cosine?  Maybe keeping the factorial underneath like math.factorial would be easier to read? Python will probably cache the value so no need to freak out about performance and it will make the code more readable.

Reply:
Yes, there are taylor expansions of cos(angle) and sin(angle)/angle.

Are you sure that python will cache them? Otherwise I could define them as constants?

Reply:
I meant explaining in the code, not to me :) 

Feel free to use constants if you want.

Reply:
I meant explaining in the code as a comment. Constants is fine for the factorial